kong/conf_loader/init.lua

```diff
@@ -660,6 +664,9 @@ local CONF_INFERENCES = {
 
   lmdb_environment_path = { typ = "string" },
   lmdb_map_size = { typ = "string" },
+
+  proxy_restricted_hosts = { typ = "array" },
+  proxy_restricted_ports = { typ = "array" },
```

Comment:
One potential use case that I can see coming up is the desire to restrict a specific <host>:<port> tuple, which would not be possible with two separate fields.

Reply:
Do you mean a single directive that accepts entries as `169.254.169.254, hostname:*, hostname:22, *:2375`? That might be interesting, what do you think @eskibars?

Reply:
Yeah, I'm pretty sure this will come up and it'd be nice to have a format that allows for it even if I can't think of any services offhand that we should default restrict with a combination host+port

Reply:
Sorry for not giving feedback on this earlier!

> `169.254.169.254, hostname:*, hostname:22, *:2375`

Yes, this is what I was suggesting. This way an operator can deny proxying to `host-a:80` and `host-b:81` while allowing proxying to `host-a:81` and `host-b:80`:

```
proxy_restricted_targets = host-a:80, host-b:81, host-c, :22
```

To me this seems like a very realistic use-case. I don't feel too strongly about it though, so if it's just me saying this, I think the feature is perfectly fine as-is.